extensions/default/src/DicomWebDataSource/retrieveStudyMetadata.js

```diff
@@ -35,8 +35,9 @@ export function retrieveStudyMetadata(
   }
 
   // Already waiting on result? Return cached promise
-  if (StudyMetaDataPromises.has(StudyInstanceUID)) {
-    return StudyMetaDataPromises.get(StudyInstanceUID);
+  const studyRetrieveId = dicomWebClient.name + StudyInstanceUID;
```

Comment:
I would not mix study instance uid with a name to create an id here. StudyMetadataPromises expects a study instance uid.

Reply:
As you can now retrieve from multiple servers, the studyinstanceuid is not a valid key anymore, so i added the client name to differentiate promisses between different servers. The map variable is defined in the file so there is no problem to change the key